### ISSUE TITLE
version `GLIBC_2.29' not found

### ISSUE THREAD

**Issue by jimafisk** [plentico/plenti#157]
Builds have been failing with our [official docker images](https://hub.docker.com/repository/docker/plentico/plenti/tags?page=1&ordering=last_updated), as detailed in [this issue](https://github.com/plentico/plenti.co/issues/11) for our public site.

Here's the error message (see [more info](https://superuser.com/questions/537683/how-to-fix-lib-x86-64-linux-gnu-libc-so-6-version-glibc-2-14-not-found)):
```
/plenti: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /plenti)
```

I think this has to due to us [recently](https://github.com/plentico/plenti/commit/cb90855472906a4d91baf3c68f3db8bd25c87d00) bumping the version of our CI builds to `ubuntu-20.04`, but our Dockerfile [still](https://github.com/plentico/plenti/blob/399085a8152bbf762a2ac6d4dbcf13b5225fdb27/Dockerfile) specifies `ubuntu-18.04`. 

I didn't realize this would matter because I figured the binary would be self-contained enough to work across versions. Since this error is related to [glibc](https://stackoverflow.com/questions/11460782/what-is-glibc-what-is-it-used-for) my guess is our CGO dependency is the root cause of this issue. For now I can probably bump our Dockerfile to `ubuntu-20.04` as well, although I might need to use an older version of libssl. Long term it would be nice if these binaries were more [portable](https://www.reddit.com/r/rust/comments/56x4r3/are_golang_binaries_more_portable_than_binaries/). Dave Cheney has some [good thoughts](https://dave.cheney.net/tag/cgo) on why cgo might be a "false economy."

**Comment by jimafisk** [comment]
Exactly our situation: https://stackoverflow.com/questions/63171297/build-application-for-ubuntu-18-from-ubuntu-20

Adding `-ldflags="-extldflags=-static"` or `-ldflags '-w -extldflags "-static"'` might solve this issue, as explained [here](https://medium.com/@diogok/on-golang-static-binaries-cross-compiling-and-plugins-1aed33499671):
> This will make the linked C part also static into the binary, reducing compatibility risk.

**Comment by jimafisk** [comment]
This might be a good opportunity to revisit trying to use `alpine` or `scratch` containers. We had a [trouble](https://github.com/plentico/plenti/issues/64) getting these to work in the past, which we attributed to [this](https://stackoverflow.com/questions/36279253/go-compiled-binary-wont-run-in-an-alpine-docker-container-on-ubuntu-host):
> By default, if using the net package a build will likely produce a binary with some dynamic linking, e.g. to libc.

They recommend setting `CGO_ENABLED=0` which we can't do, but it's possible that statically linking C could fix our problems as well.

**Comment by jimafisk** [comment]
Locally I tried `go build -ldflags="-extldflags=-static"` but that warns:
```
# github.com/plentico/plenti
/usr/bin/ld: /tmp/go-link-433344187/000032.o: in function `mygetgrouplist':
/usr/local/go/src/os/user/getgrouplist_unix.go:16: warning: Using 'getgrouplist' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /tmp/go-link-433344187/000031.o: in function `mygetgrgid_r':
/usr/local/go/src/os/user/cgo_lookup_unix.go:38: warning: Using 'getgrgid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /tmp/go-link-433344187/000031.o: in function `mygetgrnam_r':
/usr/local/go/src/os/user/cgo_lookup_unix.go:43: warning: Using 'getgrnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /tmp/go-link-433344187/000031.o: in function `mygetpwnam_r':
/usr/local/go/src/os/user/cgo_lookup_unix.go:33: warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /tmp/go-link-433344187/000031.o: in function `mygetpwuid_r':
/usr/local/go/src/os/user/cgo_lookup_unix.go:28: warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /tmp/go-link-433344187/000016.o: in function `_cgo_26061493d47f_C2func_getaddrinfo':
/tmp/go-build/cgo-gcc-prolog:58: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
```
The binary does still build, so maybe we're ok... a few sources like [this](https://stackoverflow.com/questions/2725255/create-statically-linked-binary-that-uses-getaddrinfo) and [this](https://github.com/hillu/go-yara/issues/35#issuecomment-417625806) suggest using musl instead of glibc.

**Comment by jimafisk** [comment]
This is fixed in [v0.4.15](https://github.com/plentico/plenti/releases/tag/v0.4.15) running with an `ubuntu-18.04` image. Still curious if we can get `alpine` or `scratch` to work.

**Comment by jimafisk** [comment]
Used `scratch` for the [v0.4.16 image](https://hub.docker.com/layers/plentico/plenti/v0.4.16/images/sha256-1a8bf98e75a4c45066796a877d2e0ad0ea921cc34d264d5dcefe48759077d00e?context=repo) which brings the compressed size down to `17.04 MB` from `42.51 MB` when it was on `ubuntu-18.04`. Reran the public site CI with this new container and it passed: https://github.com/plentico/plenti.co/runs/2308737699?check_suite_focus=true
